Neolution.Extensions.Identity/TokenSignManager.cs

```diff
@@ -0,0 +1,143 @@
+﻿namespace Neolution.Extensions.Identity
+{
+    using System.Security.Claims;
+    using Google.Apis.Auth;
+    using Microsoft.AspNetCore.Identity;
+    using Microsoft.Extensions.Options;
+    using Neolution.Extensions.Identity.Abstractions;
+    using Neolution.Extensions.Identity.Abstractions.Options;
+    using JsonWebToken = Neolution.Extensions.Identity.Abstractions.JsonWebToken;
+
+    /// <inheritdoc />
+    public sealed class TokenSignManager<TUser> : ITokenSignInManager<TUser>
+        where TUser : IdentityUser<Guid>
+    {
+        /// <summary>
+        /// The logger
+        /// </summary>
+        private readonly ILogger<TokenSignManager<TUser>> logger;
+
+        /// <summary>
+        /// The user manager
+        /// </summary>
+        private readonly IUserManager<TUser> userManager;
+
+        /// <summary>
+        /// The sign in manager
+        /// </summary>
+        private readonly ISignInManager<TUser> signInManager;
+
+        /// <summary>
+        /// The JWT generator
+        /// </summary>
+        private readonly IJwtGenerator<TUser> jwtGenerator;
+
+        /// <summary>
+        /// The Identity options
+        /// </summary>
+        private readonly NeolutionIdentityOptions options;
+
+        /// <summary>
+        /// Initializes a new instance of the <see cref="TokenSignManager{TUser}" /> class.
+        /// </summary>
+        /// <param name="logger">The logger.</param>
+        /// <param name="userManager">The user manager.</param>
+        /// <param name="signInManager">The sign in manager.</param>
+        /// <param name="jwtGenerator">The JWT generator.</param>
+        /// <param name="options">The Identity options.</param>
+        public TokenSignManager(ILogger<TokenSignManager<TUser>> logger, IUserManager<TUser> userManager, ISignInManager<TUser> signInManager, IJwtGenerator<TUser> jwtGenerator, IOptions<NeolutionIdentityOptions> options)
+        {
+            this.logger = logger;
+            this.userManager = userManager;
+            this.signInManager = signInManager;
+            this.jwtGenerator = jwtGenerator;
+            this.options = options.Value;
+        }
+
+        /// <inheritdoc />
+        public async Task<TUser?> PasswordSignInAsync(string username, string password)
```

Comment:
@neoscie 

there is an inconsitency between username and email. 

For us it's the same but it might not always be the case. 

So either call the parameter `email` or treat it as a `username` by calling `FindByUsernameAsync` and change the logging string

Reply:
fixed